src/fromager/sources.py

```diff
@@ -59,6 +59,13 @@ def download_source(
         ) as err:
             logger.debug(f"{req.name}: failed to resolve {req} using {url}: {err}")
             continue
+
+        # Validate version string by passing it to Version
+        try:
+            Version(str(version))
```

Comment:
Instead of initializing an object maybe using the `parse` function from `packaging.version` might be better since we are just trying to validate a version right now rather than create an object

Reply:
Okay, I will check out `parse`